js/flexbuffers.js

```diff
@@ -0,0 +1,1015 @@
+var flexbuffers = {};
+
+flexbuffers.BitWidth = {
+  WIDTH8: 0,
+  WIDTH16: 1,
+  WIDTH32: 2,
+  WIDTH64: 3,
+};
+
+flexbuffers.BitWidth.toByteWidth = (bitWidth) => {
+  return 1 << bitWidth;
+};
+
+flexbuffers.BitWidth.width = (value) => {
+  if (Number.isInteger(value)) {
+    const v = value < 0 ? value * -1 : value;
+    if (v >> 7 === 0) return flexbuffers.BitWidth.WIDTH8;
+    if (v >> 15 === 0) return flexbuffers.BitWidth.WIDTH16;
+    if (v >> 31 === 0 && v >> 32 === v) return flexbuffers.BitWidth.WIDTH32;
+    return flexbuffers.BitWidth.WIDTH64;
+  }
+
+  function _toF32(value) {
+    const buffer = new ArrayBuffer(4);
```

Comment:
2 allocations seems very inefficient just to convert a float. If anything, these buffers should be re-usable members of the builder? Is there no easier way?

Reply:
Isn't this just `Math.fround`? 🤔

Reply:
Thats exactly what I needed thanks! :)